instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/Functions.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.instrumentation.api;
+
+public final class Functions {
+
+  // TODO the majority of this can be removed/simplified if/when dropping Java 7
```

Comment:
Didn't we already agreed on dropping Java 7?

Reply:
Fixed comment, and added tracking for this TODO: #1151